src/biogeophys/SoilFluxesMod.F90

```diff
@@ -235,11 +305,27 @@ subroutine SoilFluxes (bounds, num_urbanl, filter_urbanl, &
                evaporation_demand = qflx_evap_soi(p)
                qflx_evap_soi(p)   = evaporation_limit
                qflx_ev_snow(p)    = qflx_evap_soi(p)
+               qflx_liqevap_from_top_layer(p)   = h2osoi_liq(c,j)/dtime
+               qflx_solidevap_from_top_layer(p) = h2osoi_ice(c,j)/dtime
```

Comment:
Key piece 2 of 2: As above, now we modify the partitioning directly in a way that is less sensitive to roundoff issues when h2osoi_liq and h2osoi_ice have significantly different magnitudes.